packages/dds/map/src/directory.ts

```diff
@@ -1475,8 +1482,23 @@ class SubDirectory extends TypedEventEmitter<IDirectoryEvents> implements IDirec
      * @param op - The message if from a remote delete, or null if from a local delete
      */
     private deleteSubDirectoryCore(subdirName: string, local: boolean) {
-        // This should make the subdirectory structure unreachable so it can be GC'd and won't appear in snapshots
-        // Might want to consider cleaning out the structure more exhaustively though?
-        return this._subdirectories.delete(subdirName);
+        if (this.hasSubDirectory(subdirName)) {
+            const previousValue = this.getSubDirectory(subdirName);
+            assert(previousValue !== undefined, "Subdirectory should be expected");
+            // This should make the subdirectory structure unreachable so it can be GC'd and won't appear in snapshots
+            // Might want to consider cleaning out the structure more exhaustively though?
+            const successfullyRemoved = this._subdirectories.delete(subdirName);
+            if (successfullyRemoved) {
```

Comment:
I'm not sure why it may not succeed? It looks like a Map<>, so it should always succeed - I'd rather assert that than have conditional code